### DISCUSSION TITLE
[Not converged][Print Variable]

### DISCUSSION THREAD

**Discussion by coskrrb2002** [idaholab/moose#17837] [Q&A General] [answered]
Dear moose users,

I would like to know if there is any way to print variables that did not converge during the calculation?

I want to analyze which variable causes the problem of convergence in my numerical model.

Best regrads,
Q

**Comment by simopier** [comment, marked as answer]
Hi, 

You can add the following block to your input file to see the residual for each variable of your model. 

```
[Debug]
  show_var_residual_norms = true
[../]
```

**If the residuals are very different from each other**, you might consider using scaling factors for your variables to help the convergence. It would look something like this:

```
[Variables]
  [./variable_1]
    scaling = 1e3
  [../]
  [./variable_2]
     scaling = 1
  [../]
[]
```

You might also benefit from automatic scaling. You can find information [here](https://mooseframework.inl.gov/newsletter/2020_02.html). It consists of adding these lines in your executioner block:

```
automatic_scaling = true
compute_scaling_once = false
verbose = true
```

I hope that helps!

**Reply by coskrrb2002** [reply]
Dear simopier,

It helps me a lot!

Thank you for your advice!

Best regards,
Q
